### DISCUSSION TITLE
Flac duration

### DISCUSSION THREAD

**Discussion by hazerflac7** [mifi/lossless-cut#1320] [General]
When trimming flac files the duration doesn't get updated and remains as the original. Only happens with flac audio files.

**Comment by mifi** [comment]
please provide some more info, like report from the help menu

**Comment by tordenflesk** [comment]
Same issue here. Single FLAC, split into 8 files. All play back fine but have silence equal to the original file at the end.

```
General
Complete name                            : F:\Music\_Artists\Side Brok\_Kasetter\2011 - Pina D 4\01 - Side Brok - Program A.flac
Format                                   : FLAC
Format/Info                              : Free Lossless Audio Codec
File size                                : 164 MiB
Duration                                 : 23 min 47 s
Overall bit rate mode                    : Variable
Overall bit rate                         : 962 kb/s
Album replay gain                        : -8.94 dB
Album replay gain peak                   : 0.997437
Album                                    : Pina D 4
Album/Performer                          : Pinadgreitt All-Stars
Track name                               : Program A
Track name/Position                      : 01
Performer                                : Side Brok
Recorded date                            : 2011
Cover                                    : Yes
Cover type                               : Cover (front)
Cover MIME                               : image/jpeg
COMPRESSION                              : 8

Audio
Format                                   : FLAC
Format/Info                              : Free Lossless Audio Codec
Duration                                 : 23 min 47 s
Bit rate mode                            : Variable
Bit rate                                 : 961 kb/s
Channel(s)                               : 2 channels
Channel layout                           : L R
Sampling rate                            : 44.1 kHz
Bit depth                                : 16 bits
Compression mode                         : Lossless
Replay gain                              : -9.32 dB
Replay gain peak                         : 0.997040
Stream size                              : 164 MiB (100%)
Writing library                          : libFLAC 1.3.4 (UTC 2022-02-20)
MD5 of the unencoded content             : 29F8E2E80C7F51D71E4FFE957FBD1289


```
```

No error occurred.

{
  "state": {
    "filePath": "F:\\Music\\_Artists\\Side Brok\\_Kasetter\\2011 - Pina D 4\\01 - Side Brok - Program A.flac",
    "fileFormat": "flac",
    "externalFilesMeta": {},
    "mainStreams": [
      {
        "index": 0,
        "codec_name": "flac",
        "codec_long_name": "FLAC (Free Lossless Audio Codec)",
        "codec_type": "audio",
        "codec_tag_string": "[0][0][0][0]",
        "codec_tag": "0x0000",
        "sample_fmt": "s16",
        "sample_rate": "44100",
        "channels": 2,
        "channel_layout": "stereo",
        "bits_per_sample": 0,
        "r_frame_rate": "0/0",
        "avg_frame_rate": "0/0",
        "time_base": "1/44100",
        "start_pts": 0,
        "start_time": "0.000000",
        "duration_ts": 62961231,
        "duration": "1427.692313",
        "bits_per_raw_sample": "16",
        "extradata_size": 34,
        "disposition": {
          "default": 0,
          "dub": 0,
          "original": 0,
          "comment": 0,
          "lyrics": 0,
          "karaoke": 0,
          "forced": 0,
          "hearing_impaired": 0,
          "visual_impaired": 0,
          "clean_effects": 0,
          "attached_pic": 0,
          "timed_thumbnails": 0,
          "captions": 0,
          "descriptions": 0,
          "metadata": 0,
          "dependent": 0,
          "still_image": 0
        },
        "side_data_list": [
          {
            "side_data_type": "Replay Gain"
          }
        ]
      },
      {
        "index": 1,
        "codec_name": "mjpeg",
        "codec_long_name": "Motion JPEG",
        "profile": "Baseline",
        "codec_type": "video",
        "codec_tag_string": "[0][0][0][0]",
        "codec_tag": "0x0000",
        "width": 600,
        "height": 600,
        "coded_width": 600,
        "coded_height": 600,
        "closed_captions": 0,
        "film_grain": 0,
        "has_b_frames": 0,
        "sample_aspect_ratio": "1:1",
        "display_aspect_ratio": "1:1",
        "pix_fmt": "yuvj420p",
        "level": -99,
        "color_range": "pc",
        "color_space": "bt470bg",
        "chroma_location": "center",
        "refs": 1,
        "r_frame_rate": "90000/1",
        "avg_frame_rate": "0/0",
        "time_base": "1/90000",
        "start_pts": 0,
        "start_time": "0.000000",
        "duration_ts": 128492308,
        "duration": "1427.692311",
        "bits_per_raw_sample": "8",
        "disposition": {
          "default": 0,
          "dub": 0,
          "original": 0,
          "comment": 0,
          "lyrics": 0,
          "karaoke": 0,
          "forced": 0,
          "hearing_impaired": 0,
          "visual_impaired": 0,
          "clean_effects": 0,
          "attached_pic": 1,
          "timed_thumbnails": 0,
          "captions": 0,
          "descriptions": 0,
          "metadata": 0,
          "dependent": 0,
          "still_image": 0
        },
        "tags": {
          "comment": "Cover (front)"
        }
      }
    ],
    "copyStreamIdsByFile": {
      "F:\\Music\\_Artists\\Side Brok\\_Kasetter\\2011 - Pina D 4\\01 - Side Brok - Program A.flac": {
        "0": true,
        "1": true
      }
    },
    "cutSegments": [
      {
        "end": 264.3547814881926
      },
      {
        "start": 264.3547814881926,
        "end": 453.54166225706865
      },
      {
        "start": 453.54166225706865,
        "end": 648.1962045340167
      },
      {
        "start": 648.1962045340167,
        "end": 774.4420740575948
      },
      {
        "start": 774.4420740575948,
        "end": 869.8457104255934
      },
      {
        "start": 869.8457104255934,
        "end": 1094.5722339911622
      },
      {
        "start": 1094.5722339911622,
        "end": 1126.500282152672
      },
      {
        "start": 1126.500282152672
      }
    ],
    "mainFileFormatData": {
      "filename": "F:\\Music\\_Artists\\Side Brok\\_Kasetter\\2011 - Pina D 4\\01 - Side Brok - Program A.flac",
      "nb_streams": 2,
      "nb_programs": 0,
      "format_name": "flac",
      "format_long_name": "raw FLAC",
      "start_time": "0.000000",
      "duration": "1427.692313",
      "size": "171625543",
      "bit_rate": "961694",
      "probe_score": 100,
      "tags": {
        "Album": "Pina D 4",
        "Artist": "Side Brok",
        "Date": "2011",
        "Title": "Program A",
        "track": "01",
        "replaygain_album_gain": "-8.94 dB",
        "replaygain_album_peak": "0.997437",
        "replaygain_track_gain": "-9.32 dB",
        "replaygain_track_peak": "0.997040",
        "album_artist": "Pinadgreitt All-Stars",
        "COMPRESSION": "8"
      }
    },
    "rotation": 360,
    "shortestFlag": false,
    "effectiveExportMode": "separate"
  },
  "platform": "win32",
  "version": "3.46.2"
}
```

**Comment by tordenflesk** [comment]
re-encoding with ffmpeg reports:
ERROR: 2 consecutive FLAC__STREAM_DECODER_END_OF_STREAM events

**Comment by mifi** [comment]
Might be caused by the mjpeg track. try to disable it and see if it works

**Reply by slrslr** [reply]
> Might be caused by the mjpeg track. try to disable it and see if it works

I have this issue too @mifi . My source file does not contain mjpeg track, only flac, 2c stereo track.
[Here i am sending source file and .llc file.](https://drive.proton.me/urls/0R73RZH5PM#GzDTXE1OLZPg)
Reproduce:
1) export (no matter which cut type and negative ts you select)
2) play file in MPV or VLC (I am on Linux) - you can see original length (wrong, something to be fixed), not cut start to cut end.
The file play well from cut start, but when seeking beyond cut end, it resets the playback from start.

**Reply by mifi** [reply]
thanks for sharing the file. Indeed I can reproduce the problem.
I can even reproduce with this simplified ffmpeg command:

```bash
ffmpeg -i 'lossless-cut-issue(discussion)-1320/GTR (Gareth Emery) - Mistral (Original Mix).flac' -t 10 -c copy -y out.flac
```
even with a 10 sec `-t` argument the output duration is still 9min 22sec. So it must be a bug in ffmpeg.
Maybe someone can test with ffmpeg 6 - i'm still waiting for homebrew support for ffmpeg 6: https://github.com/Homebrew/homebrew-core/pull/124363

If it still doesn't work on ffmpeg 6, someone should submit a bug report at https://trac.ffmpeg.org/

**Reply by slrslr** [reply]
I did:
`cd /usr/local/bin && sudo wget -c https://johnvansickle.com/ffmpeg/builds/ffmpeg-git-$(dpkg --print-architecture)-static.tar.xz && sudo tar -xf ffmpeg-git-* && cd ffmpeg-git-*static && ./ffmpeg -version`
ffmpeg version N-65014-g7c130d6911-static
(should be built on 2023 03 13 https://johnvansickle.com/ffmpeg/)

Then ran your ffmpeg command:
`./ffmpeg -i '/home/user/Downloads/GTR (Gareth Emery) - Mistral (Original Mix).flac' -t 10 -c copy -y out.flac`
and play the out.flac -> duration is wrong (same as original file) and moreover unable to seek.

So i have used lossless-cut to cut part of the source flac file using provided llc file, export it, then from Tools menu selected "Last ffmpeg commands" menu entry, copied ffmpeg command:
`ffmpeg -hide_banner -ss '111.17848' -i '/home/user/Downloads/GTR (Gareth Emery) - Mistral (Original Mix).flac' -t '369.79157' -avoid_negative_ts make_zero -map '0:0' '-c:0' copy -map_metadata 0 -movflags '+faststart' -default_mode infer_no_subs -ignore_unknown -f flac -y '/dev/shm/GTR (Gareth Emery) - Mistral (Original Mix)-00.01.51.178-00.08.00.970.flac'`

 and replaced ffmpeg in it by that ffmpeg binary downloaded above. Output file has same issue as older ffmpeg version.
# There is issue https://trac.ffmpeg.org/ticket/10287

**Reply by mifi** [reply]
here's their docuemntation for reporting bugs:
https://ffmpeg.org/bugreports.html

i can do it when i have some time but very busy now

**Reply by slrslr** [reply]
Hello @mifi the FFmpeg issue/ticket linked and highlighted just before your comment, was just updated and marked invalid. Can you please comment that issue on the tracker? I think that the LC is producing the command that is mentioned inside the issue.

**Reply by mifi** [reply]
maybe you can ask if there is any way to change the stream duration when outputting flac

**Reply by Slavomir9** [reply]
Hello,
on my Linux Ubuntu work correct this command
`ffmpeg -ss 00:00:20 -to 00:00:40 -i input.flac output.flac`

**Reply by mifi** [reply]
@Slavomir9 works, as in it outputs the correct duration? which version of ffmpeg?

so if you try the below command then it doesn't work, but your command it works?

```
ffmpeg -hide_banner -ss '111.17848' -i 'input.flac' -t '369.79157' -avoid_negative_ts make_zero -map '0:0' '-c:0' copy -map_metadata 0 -movflags '+faststart' -default_mode infer_no_subs -ignore_unknown -f flac -y 'output.flac'
```

EDIT: nevermind! you're not using `-c copy` so you're re-encoding.

**Reply by Slavomir9** [reply]
@mifi Yes. It is the same as you speak. Longer command not work for me. And shorter command work correct.
My version of FFMPEG is:
**`n4.4.4-6-gd5fa6e3a91`**

**Reply by mifi** [reply]
I edited my comment. basically it works with your short command because you're re-encoding, which is much slower than `-copy`

**Reply by Slavomir9** [reply]
@mifi  My short command is lossless. FLAC is lossless format. FLAC work on different principle as losing audio formats (m4a, mp3 or Vorbis for example) on the same command. I use extremely old computer processor from 2007 and on my short command work extremely fast with low activity of processor

**Reply by mifi** [reply]
Technically lossless yes, but it’s not a stream copy, so if you have a large file it’s going to be slower than copy. But maybe we could still do it when cutting files that only have a single flac stream, because the alternative is a broken output flac file.

**Reply by mifi** [reply]
Created an issue #1809

**Reply by Slavomir9** [reply]
@mifi OK, thanks.
Yes. I tested the short command on mono (non-stereo files)

**Comment by slrslr** [comment]
**Workaround** ("sox" Linux utility - Slavomir9 mentioned it):
1) Export .flac file segments using LoslessCut. Btw. it exported with .mp4 extension despite "flac - raw FLAC" output format (container) was select, so i have renamed files to .flac. [Resulting files are here if anyone is interested](https://drive.proton.me/urls/ABDECRF2DG#jL2GRlf4J6n4).
2) use Linux utility sox to merge segments: sox a01.flac a02.flac a03.flac a04.flac a.flac
Resulting duration is correct, unlike using LosslessCut merging (where it shows duration of original file). File sizes are the same in both cases.
